### PR TITLE
SVD++ demo

### PR DESCRIPTION
**Pull request by jdk8**

### CODE REVIEW

**Comment by jegonzal** [comment]
This is a very useful application, thanks for the PR!  Two requests:

1) Can you make a version that takes the edges RDD (or even a graph) instead of the raw text file.
2) Can you add a basic unit test that generates a synthetic edge RDD and then runs the inference algorithm.  For a reference collection of unit tests see https://github.com/amplab/graphx/blob/master/graph/src/test/scala/org/apache/spark/graph/AnalyticsSuite.scala

By doing this we can make the code more general and also have a way to verify that changes we make to the system don't affect the application.

**Comment by jdk8** [comment]
Hi jegonzal, got it, I will make these changes later.

**Comment by ankurdave** [comment]
Jenkins, add to whitelist.

File: graph/src/main/scala/org/apache/spark/graph/algorithms/Svdpp.scala [hunk 1]
```diff
@@ -0,0 +1,158 @@
+package org.apache.spark.graph.algorithms
+
+import org.apache.spark._
+import org.apache.spark.rdd._
+import org.apache.spark.graph._
+import scala.util.Random
+import org.apache.commons.math.linear._
+
+class VD (
```
Comment (jegonzal) [inline graph/src/main/scala/org/apache/spark/graph/algorithms/Svdpp.scala]: Can you rename `VD` to `VertexFactor` or something slightly more descriptive?  We were using `VD` as a type parameter.

**Comment by jegonzal** [comment]
This looks great!  Thanks for the contribution.
